src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java

```diff
@@ -0,0 +1,68 @@
+package mate.academy.dao.impl;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.MovieSessionDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.MovieSession;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieSessionDaoImpl extends AbstractDao implements MovieSessionDao {
+    public MovieSessionDaoImpl() {
+        super(HibernateUtil.getSessionFactory());
+    }
+
+    @Override
+    public MovieSession add(MovieSession movieSession) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(movieSession);
+            transaction.commit();
+            return movieSession;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add a movie session: " + movieSession, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public Optional<MovieSession> get(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            return Optional.ofNullable(session.get(MovieSession.class, id));
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get a movie session with id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        try (Session session = sessionFactory.openSession()) {
+            Query<MovieSession> getAvailableSessionsQuery = session.createQuery(
+                    "from MovieSession ms "
+                            + "where ms.movie.id = :movie_id and "
+                            + "ms.showTime between :start_time and :end_time",
+                    MovieSession.class);
+            getAvailableSessionsQuery.setParameter("movie_id", movieId);
+            getAvailableSessionsQuery.setParameter("start_time", date.atTime(0, 0, 0));
+            getAvailableSessionsQuery.setParameter("end_time", date.atTime(23, 59, 59));
```

Comment:
you can use LocalTime.MIN and LocalTime.MAX. also for parameters usually  used  camelCase